httomo/wrappers_class.py

```diff
@@ -0,0 +1,191 @@
+
+    def _execute_reconstruction(self, 
+                               method_name: str,
+                               params: Dict,
+                               data: xp.ndarray,
+                               angles_radians: np.ndarray) -> xp.ndarray:
+        """The reconstruction wrapper.
+
+        Args:
+            method_name (str): The name of the method to use.
+            params (Dict): A dict containing independent of httomo params.
+            data (xp.ndarray): a numpy or cupy data array.
+            angles_radians (np.ndarray): a numpy array of projection angles.
+
+        Returns:
+            xp.ndarray: a numpy or cupy array of the reconstructed data.
+        """
+        # for 360 degrees data the angular dimension will be truncated while angles are not.
+        # Truncating angles if the angular dimension has got a different size
+        angular_dim_size = np.size(data, 0)
+        if np.size(data, 0) != len(angles_radians):
+            angles_radians = angles_radians[0:angular_dim_size]
+                
+        data = getattr(self.module, method_name)(data, angles_radians, **params)
+        return data
+
+    def _execute_rotation(self, 
+                          method_name: str,
+                          params: Dict,
+                          data: xp.ndarray) -> float:
+        """The center of rotation wrapper.
+
+        Args:
+            method_name (str): The name of the method to use.
+            params (Dict): A dict containing independent of httomo params.
+            data (xp.ndarray): a numpy or cupy data array.
+
+        Returns:
+            float: The center of rotation.
+        """
+        rot_center = 0
+        mid_rank = int(round(self.comm.size / 2) + 0.1)
+        if self.comm.rank == mid_rank:
+            if params['ind'] == 'mid':
+                params['ind'] = data.shape[1] // 2 # get the middle slice
+            rot_center = getattr(self.module, method_name)(data, **params)
+        rot_center = self.comm.bcast(rot_center, root=mid_rank)
+        return rot_center
+    
+class TomoPyWrapper(BaseWrapper):
+    """A class that wraps TomoPy functions for httomo
+    """
+    def __init__(self,
+                 module_name: str,
+                 function_name: str,
+                 method_name: str,
+                 comm: Comm):
+        super().__init__(module_name, function_name, method_name, comm)
+        
+        # if not changed bellow the generic wrapper will be executed
+        self.wrapper_method = super(TomoPyWrapper, self)._execute_generic
+
+        if module_name in ["misc", "prep", "recon"]:
+            from importlib import import_module
+            self.module = getattr(import_module("tomopy." + module_name), function_name)
+            if function_name == "normalize":
+                func = getattr(self.module, method_name)
+                sig_params = signature(func).parameters
+                if 'dark' in sig_params and 'flat' in sig_params:
```

Comment:
```suggestion
                if 'darks' in sig_params and 'flats' in sig_params:
```

Reply:
TomoPy [uses](https://tomopy.readthedocs.io/en/stable/api/tomopy.prep.normalize.html#tomopy.prep.normalize.normalize) `dark` and `flat` naming somehow.